### PR TITLE
Add record to hotspot stats when packets come in

### PR DESCRIPTION
**Pull request by ambuznego**

### CODE REVIEW

File: lib/console_web/controllers/router/device_controller.ex [hunk 1]
```diff
@@ -241,6 +242,25 @@ defmodule ConsoleWeb.Router.DeviceController do
...
+                "channel" => event.data["hotspot"]["channel"],
+                "spreading" => event.data["hotspot"]["spreading"],
+                "category" => event.category,
+                "sub_category" => event.sub_category,
+                "reported_at_epoch" => event.reported_at_epoch,
+                "device_id" => device.id,
+                "organization_id" => device.organization_id
+              })
+            else
+              {:ok, %{}}
```
Comment (vicmgs) [inline lib/console_web/controllers/router/device_controller.ex:261]: this line was funny for some reason LOL

File: lib/console_web/controllers/router/device_controller.ex [hunk 2]
```diff
@@ -241,6 +242,25 @@ defmodule ConsoleWeb.Router.DeviceController do
               {:ok, %{}}
             end
           end)
+          |> Ecto.Multi.run(:hotspot_stat, fn _repo, %{ event: event, device: device } ->
+            if event.sub_category in ["uplink_confirmed", "uplink_unconfirmed", "join_request"] do
```
Comment (vicmgs) [inline lib/console_web/controllers/router/device_controller.ex:246]: join_request is not a sub_category, its the category. Join_request's sub_category is actually undefined lol

File: lib/console_web/controllers/router/device_controller.ex (hunk 1 above)
Comment (ambuznego) [reply lib/console_web/controllers/router/device_controller.ex:261]: I'm humorous like that

File: lib/console_web/controllers/router/device_controller.ex (hunk 2 above)
Comment (ambuznego) [reply lib/console_web/controllers/router/device_controller.ex:246]: you is correct

File: lib/console_web/controllers/v1/alert_controller.ex [hunk 3]
```diff
@@ -29,7 +28,7 @@ defmodule ConsoleWeb.V1.AlertController do
       nil ->
         {:error, :not_found, "Alert not found"}
       %Alert{} = alert ->
-        with {:ok, %Alert{} = alert} <- Alerts.update_alert(alert, alert_params) do
+        with {:ok, %Alert{} = _alert} <- Alerts.update_alert(alert, alert_params) do
```
Comment (vicmgs) [inline lib/console_web/controllers/v1/alert_controller.ex:31]: if we don't need the variable, we can just with {:ok, %Alert{} }

File: lib/console_web/controllers/v1/alert_controller.ex [hunk 4]
```diff
@@ -59,7 +58,7 @@ defmodule ConsoleWeb.V1.AlertController do
         nil ->
           {:error, :not_found, "Alert not found"}
         %Alert{} = alert ->
-          with {:ok, %AlertNode{} = alert_node} <- Alerts.add_alert_node(current_organization, alert, node_id, node_type) do
+          with {:ok, %AlertNode{} = _alert_node} <- Alerts.add_alert_node(current_organization, alert, node_id, node_type) do
```
Comment (vicmgs) [inline lib/console_web/controllers/v1/alert_controller.ex:61]: same here

File: lib/console_web/controllers/router/device_controller.ex (hunk 2 above)
Comment (ambuznego) [reply lib/console_web/controllers/router/device_controller.ex]: fixed

**Review by vicmgs — APPROVED** [review]
naiiiisuu

**Pull request merged**